File: pset1/greedy.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

/*Variables*/
    float change;
    int changeleft = 0;
    int coincount = 0;


int main(void)
{
    /*Query on change*/
    do
    {
    printf("How much do you owe? (input in (X.XX))\n");
    change = get_float();
    } while (change < 0);
    
    /*Change from dollar to cents*/
    changeleft = round(change * 100);
    
    /*Calculate coin required*/
    while (changeleft >= 25)
    {
      coincount++;
      changeleft -= 25;
    }
    
    while (changeleft >= 10)
    {
      coincount++;
      changeleft -= 10;
    }
    
    while (changeleft >= 5)
    {
      coincount++;
      changeleft -= 5;
    }
    
    while (changeleft >= 1)
    {
      coincount++;
      changeleft -= 1;
    }
    
    /*Print coin required*/
    printf("%i\n", coincount);
}

File: pset6/greedy.py
import cs50

def main():
    changeleft = check() * 100
    count = 0
    while changeleft >= 25:
        count += 1
        changeleft -= 25
        
    while changeleft >= 10:
        count += 1
        changeleft -= 10
        
    while changeleft >= 5:
        count += 1
        changeleft -= 5
        
    while changeleft >= 1:
        count += 1
        changeleft -= 1
        
    print("{}".format(count))
    
def check():
    while True:
        print("O hai! How much change is owed?")
        change = cs50.get_float()
        if change > 0:
            break
    return change
    
if __name__ == "__main__":
    main()